src/test/regress/input/multi_complex_count_distinct.source

```diff
@@ -0,0 +1,129 @@
+--
+-- COMPLEX_COUNT_DISTINCT
+--
+
+CREATE TABLE lineitem_hash (
+	l_orderkey bigint not null,
+	l_partkey integer not null,
+	l_suppkey integer not null,
+	l_linenumber integer not null,
+	l_quantity decimal(15, 2) not null,
+	l_extendedprice decimal(15, 2) not null,
+	l_discount decimal(15, 2) not null,
+	l_tax decimal(15, 2) not null,
+	l_returnflag char(1) not null,
+	l_linestatus char(1) not null,
+	l_shipdate date not null,
+	l_commitdate date not null,
+	l_receiptdate date not null,
+	l_shipinstruct char(25) not null,
+	l_shipmode char(10) not null,
+	l_comment varchar(44) not null,
+	PRIMARY KEY(l_orderkey, l_linenumber) );
+	
+SELECT master_create_distributed_table('lineitem_hash', 'l_orderkey', 'hash');
+SELECT master_create_worker_shards('lineitem_hash', 8, 1);
+
+\COPY lineitem_hash FROM '@abs_srcdir@/data/lineitem.1.data' with delimiter '|'
+\COPY lineitem_hash FROM '@abs_srcdir@/data/lineitem.2.data' with delimiter '|'
+
+SET citus.task_executor_type to "task-tracker";
+
+-- count(distinct) is supported on top level query if there
+-- is a grouping on the partition key	
+SELECT 
+	l_orderkey, count(DISTINCT l_partkey) 
+	FROM lineitem_hash 
+	GROUP BY l_orderkey 
+	ORDER BY 2 DESC 
+	LIMIT 10;
+
+-- it is not supported if there is no grouping or grouping is on non-partition field
+SELECT 
+	count(DISTINCT l_partkey) 
+	FROM lineitem_hash 
+	ORDER BY 1 DESC 
+	LIMIT 10;
+
+SELECT 
+	l_shipmode, count(DISTINCT l_partkey) 
+	FROM lineitem_hash 
+	GROUP BY l_shipmode 
+	ORDER BY 2 DESC 
+	LIMIT 10;
+
+-- count distinct is supported on single table subqueries
+SELECT *
+	FROM (
+		SELECT 
+			l_orderkey, count(DISTINCT l_partkey) 
+			FROM lineitem_hash 
+			GROUP BY l_orderkey) sub
+	ORDER BY 2 DESC 
```

Comment:
Do you think should we add `ORDER BY` on both target entries to ensure that the results are always same? There are similar tests below as well.